app/src/main/java/com/everest/userinfo/EditActivity.kt

```diff
@@ -0,0 +1,165 @@
+        outState.putBoolean(PHONE_NUMBER, binding.phoneNumberET.isEnabled)
+        outState.putBoolean(PINCODE, binding.pincodeET.isEnabled)
+        outState.putBoolean(ADDRESS, binding.addressET.isEnabled)
+        outState.putString(HEADER_TEXT, binding.headerTV.text.toString())
+        super.onSaveInstanceState(outState)
+    }
+
+    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
+        binding.validateButton.visibility = savedInstanceState.getInt(VALIDATE_BUTTON)
+        binding.cancelButton.visibility = savedInstanceState.getInt(CANCEL_BUTTON)
+        binding.confirmButton.visibility = savedInstanceState.getInt(CONFIRM_BUTTON)
+        isEnabled = savedInstanceState.getBoolean(USERNAME)
+        if (!isEnabled) {
+            customizeEditTextToDisplayEnteredDetails(binding.layout)
+        }
+        binding.usernameET.isEnabled = savedInstanceState.getBoolean(USERNAME)
+        binding.emailET.isEnabled = savedInstanceState.getBoolean(EMAIL)
+        binding.phoneNumberET.isEnabled = savedInstanceState.getBoolean(PHONE_NUMBER)
+        binding.pincodeET.isEnabled = savedInstanceState.getBoolean(PINCODE)
+        binding.addressET.isEnabled = savedInstanceState.getBoolean(ADDRESS)
+
+        binding.headerTV.text = savedInstanceState.getString(HEADER_TEXT)
+        super.onRestoreInstanceState(savedInstanceState)
+    }
+
+    private fun customizeEditTextToDisplayEnteredDetails(layout: ConstraintLayout) {
+        for (i in 0 until layout.childCount) {
+            val v = layout.getChildAt(i)
+            if (v is EditText) {
+                v.isEnabled = false
+                v.setBackgroundResource(0)
+                v.setPadding(2, 2, 2, 2)
+                v.setTextColor(Color.parseColor("#000000"))
+            }
+        }
+    }
+
+    private fun validateData() {
+        if (areFieldsEmpty(binding.layout) && isEmailValid(
+                binding.emailET.text.toString(),
+            ) && isPhoneNumberValid(
+                binding.phoneNumberET.text.toString()
+            ) && isPinCodeValid(binding.pincodeET.text.toString())
+        ) {
+            confirmDetails()
+        }
+    }
+
+    private fun confirmDetails() {
+        customizeEditTextToDisplayEnteredDetails(binding.layout)
+        binding.confirmButton.visibility = View.VISIBLE
+        binding.cancelButton.visibility = View.VISIBLE
+        binding.validateButton.visibility = View.INVISIBLE
+        binding.headerTV.text = DETAILS_HEADER_TEXT
+    }
+
+    private fun editDetails(layout: ConstraintLayout) {
+        for (i in 0 until layout.childCount) {
+            val v = layout.getChildAt(i)
+            if (v is EditText) {
+                v.isEnabled = true
+                v.setBackgroundResource(R.drawable.rounded_border_edittext)
+                v.setPadding(40, 45, 45, 40)
+            }
+            binding.addressET.setPadding(40, 45, 45, 170)
+            binding.cancelButton.visibility = View.INVISIBLE
```

Comment:
You can check `Group` to handle multiple views visibility when using ConstraintLayout
https://developer.android.com/reference/androidx/constraintlayout/widget/Group